sarif/sarif.go

```diff
@@ -346,6 +360,84 @@ func (s *SarifTransformer) isSnykURI(uri string) bool {
 	return strings.HasPrefix(uri, "https_//")
 }
 
+func (s *SarifTransformer) relativePath(path string) (string, error) {
+	if s.workspacePath == "" {
+		return path, nil
+	}
+
+	if !strings.HasPrefix(path, s.workspacePath) {
+		return "", errors.Errorf(
+			"%s: result is not inside expected directory: %s",
+			path,
+			s.workspacePath,
+		)
+	}
+
+	relativePath, err := filepath.Rel(s.workspacePath, path)
+	if err != nil {
+		return "", errors.Errorf(
+			"could not get relative path from path %s using prefix %q",
+			path,
+			s.workspacePath,
+		)
+	}
+
+	return relativePath, nil
+}
+
+// normalisePath will take a given path and construct a file url pointing to
+// the file relative to the workspacePath
+func (s *SarifTransformer) normalisePath(
+	path string,
+	uriBaseId *string,
+) (*ocsf.File, error) {
+	parsedPath, err := url.Parse(path)
+	if err != nil {
+		return nil, errors.Errorf("%s: could not parse path: %w", path, err)
+	}
+
+	cleanedPath := filepath.Clean(
+		filepath.Join(parsedPath.Host, parsedPath.Path),
+	)
+
+	switch {
+	case uriBaseId == nil:
+		if s.workspacePath != "" && filepath.IsAbs(cleanedPath) {
+			relativePath, err := s.relativePath(cleanedPath)
+			if err != nil {
+				return nil, err
+			}
+
+			cleanedPath = relativePath
+		}
+	case strings.ToLower(*uriBaseId) == "%srcroot%" && filepath.IsAbs(cleanedPath):
+		// this should be a relative path and it's not, so we should return an error
+		return nil, errors.Errorf("%s: path was expected to be relative but it's absolute", cleanedPath)
+	case strings.ToLower(*uriBaseId) == "rootpath":
```

Comment:
same here, rootpath is trivy specific, for trivy this will always return an error